src/ApiCallable.php

```diff
@@ -111,12 +112,34 @@ private static function setPageStreaming($callable, $pageStreamingDescriptor)
         return $inner;
     }
 
+    private static function setCustomHeader($callable, $headerDescriptor)
+    {
+        $inner = function() use ($callable, $headerDescriptor) {
+            $params = func_get_args();
+            if (count($params) != self::GRPC_CALLABLE_PARAM_COUNT ||
+                gettype($params[self::GRPC_CALLABLE_METADATA_INDEX]) != 'array') {
```

Comment:
I know this pattern was already used above, but I think it would be better to use the built-in function is_array (http://php.net/manual/en/function.is-array.php)